content/python/suspend_virtual_server.py.md

```diff
@@ -0,0 +1,364 @@
+
+filter_location = {'regions': {'keyname': {'operation': location}}}
+
+location_list = client["SoftLayer_Location"].getDatacenters(mask='mask[id,name,regions[keyname]]',
+                                                            filter=filter_location)
+locationId = location_list[0]['id']
+
+package_list = client["SoftLayer_Product_Package"].getAllObjects(filter=filter_package)
+packageId = package_list[0]['id']
+
+preset_list = client["SoftLayer_Product_Package"].getActivePresets(mask='mask[prices[item]]',
+                                                                   filter=filter_preset, id=packageId)
+presetId = preset_list[0]['id']
+
+item_prices = client["SoftLayer_Product_Package"].getItemPrices(mask='mask[item]', id=packageId)
+
+capacity = 0
+for preset in preset_list:
+    for price in preset['prices']:
+        if "CORE" in price['item']['keyName']:
+            capacity = price['item']['capacity']
+
+prices = []
+for keyname_item in items:
+    for item_price in item_prices:
+        if keyname_item == item_price['item']['keyName']:
+            if 'capacityRestrictionMinimum' in item_price:
+                capacityMinimun = item_price['capacityRestrictionMinimum']
+                capacityMaximum = item_price['capacityRestrictionMaximum']
+                if capacityMinimun >= capacity and capacityMaximum <= capacity:
+                    prices.append(item_price['id'])
+            else:
+                prices.append(item_price['id'])
+
+print(' '.join(map(str, prices)))
+
+orderData = {
+    "complexType": "SoftLayer_Container_Product_Order_Virtual_Guest",
+    'virtualGuests': [
+        {
+            'hostname': 'SuspendVsTest',
+            'domain': 'softlayer.test'
+        }
+    ],
+    "useHourlyPricing": True,
+    'packageId': packageId,
+    'location': locationId,
+    'presetId': presetId,
+    'prices': [{'id': price_id} for price_id in prices],
+    'quantity': 1
+}
+
+try:
+    # Order a virtual guest Suspend.
+    response = client['SoftLayer_Product_Order'].placeOrder(orderData)
+    print(json.dumps(response, sort_keys=False, indent=4, separators=(',', ': ')))
+
+except SoftLayer.SoftLayerAPIError as e:
+    """ 
+        If there was an error returned from the SoftLayer API then bomb out with the 
+        error message. 
+        """
+    print("Unable to order a VSI \nfaultCode= %s, \nfaultString= %s"
+          % (e.faultCode, e.faultString))
+
+```
+
+###Create an Image Template from a Suspend Virtual Server
+
+```python
+import json
+
+import SoftLayer
+
+# Declare the API client
+client = SoftLayer.create_client_from_env()
+
+virtualGuestId = 11111
+
+disk_capacity = 100
+
+block_devices = client['SoftLayer_Virtual_Guest'].getBlockDevices(mask='mask[diskImage]', id=virtualGuestId)
+
+deviceId = 0
+for device in block_devices:
+    if disk_capacity == device['diskImage']['capacity']:
+        deviceId = device['id']
+
+groupName = "test suspend vs"
+
+blockDevices = [
```

Comment:
this can be condensed onto one line for readability. 
`blockDevices = [{'id': deviceId}]`

Reply:
Done